src/components/common/YourCompany/StartDate.vue

```diff
@@ -0,0 +1,226 @@
+          </div>
+          <div class="float-right mb-2">
+            <v-btn large color="primary" id="start-done-btn" class="mr-2" @click="onDoneClicked()">
+              <span>Done</span>
+            </v-btn>
+            <v-btn large outlined color="primary" id="start-cancel-btn" @click="onCancelClicked()">
+              <span>Cancel</span>
+            </v-btn>
+          </div>
+        </template>
+
+        <span class="mt-n2 mr-n3 float-right">
+          <!-- Correct changes button -->
+          <v-btn v-if="isCorrectionFiling && !onEditMode && !isCorrected" text color="primary"
+            id="start-changes-btn" @click="onChangeClicked()">
+            <v-icon small color="primary">mdi-pencil</v-icon>
+            <span>{{ editLabel }}</span>
+          </v-btn>
+
+          <!-- Undo changes buttons -->
+          <v-btn v-if="isCorrectionFiling && !onEditMode && isCorrected"
+          text color="primary" id="start-undo-btn" @click="onUndoClicked()">
+            <v-icon small>mdi-undo</v-icon>
+            <span>Undo</span>
+          </v-btn>
+          <v-menu v-if="isCorrectionFiling && !onEditMode && isCorrected"
+            offset-y left nudge-bottom="4" v-model="dropdown">
+            <template v-slot:activator="{ on }">
+              <v-btn text small color="primary" id="start-menu-btn" v-on="on">
+                <v-icon>{{dropdown ? 'mdi-menu-up' : 'mdi-menu-down'}}</v-icon>
+              </v-btn>
+            </template>
+            <v-btn text color="primary" id="more-changes-btn"
+              @click="onChangeClicked(); dropdown = false">
+              <v-icon small color="primary">mdi-pencil</v-icon>
+              <span>{{ editLabel }}</span>
+            </v-btn>
+          </v-menu>
+        </span>
+
+      </v-col>
+    </v-row>
+  </section>
+</template>
+
+<script lang="ts">
+import { Component, Mixins, Prop, Watch } from 'vue-property-decorator'
+import { Getter, Action } from 'vuex-class'
+import { CommonMixin, DateMixin } from '@/mixins/'
+import { DatePicker as DatePickerShared } from '@bcrs-shared-components/date-picker'
+import { ActionBindingIF } from '@/interfaces'
+
+@Component({
+  components: {
+    DatePickerShared
+  }
+})
+export default class StartDate extends Mixins(CommonMixin, DateMixin) {
+  // Global getters
+  @Getter isBenIaCorrectionFiling!: boolean
+  @Getter isFirmCorrectionFiling!: boolean
+  @Getter getCorrectionStartDate!: string
+  @Getter getBusinessFoundingDate!: string // actually date-time
+  @Getter getCurrentDate!: string
+  @Getter getCurrentJsDate!: Date
+
+  // Global setter
+  @Action setCorrectionStartDate!: ActionBindingIF
+  @Action setValidComponent!: ActionBindingIF
+
+  /** Whether to show invalid section styling. */
+  @Prop({ default: false })
+  readonly invalidSection: boolean
+
+  protected onEditMode = false
+  protected isCorrected = !!this.getCorrectionStartDate
+  protected newCorrectedStartDate = undefined // date is "Month Day, Year"
```

Comment:
It's risky to initialize properties as undefined because Vue won't make them reactive (in the template or getters). A better choice here would be
```
protected newCorrectedStartDate = null as string
```

Or you could initialize it to `''` because an empty string is falsy.

Reply:
Okay I will change that to null as string

Reply:
I think this is indeed part of your problem.
1. initialize startDate to null in state-models.ts
2. do not initialize/reset variables to undefined in StartDate.vue

I would add console.logs in various places to verify that variables are being set and read (get) correctly.